### PR TITLE
Add composition in Relation.Binary.Construct

### PR DESCRIPTION
**Pull request by calintat**
Hi, I thought it would be useful to have composition of binary relations in Relation.Binary.Construct since I needed it in one of my projects. I also proved a few properties about composition (I tried to be consistent with the other Relation.Binary.Construct modules). Let me know what you think.

### CODE REVIEW

File: src/Relation/Binary/Construct/Composition.agda [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+--
+-- Composition of two binary relations
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Relation.Binary.Construct.Composition where
+
+open import Data.Product
+open import Function
```
Comment (MatthewDaggitt) [inline src/Relation/Binary/Construct/Composition.agda:12]: You can get away with importing `Function.Base`.

File: src/Relation/Binary/Construct/Composition.agda [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  variable
+    a b c ℓ ℓ₁ ℓ₂ : Level
+    A : Set a
+    B : Set b
+    C : Set c
+
+------------------------------------------------------------------------
+-- Definition
+
+_;_ : ∀ {a b c ℓ₁ ℓ₂} {A : Set a} {B : Set b} {C : Set c} →
```
[the remark this answers is not in the document]
Comment (calintat) [reply src/Relation/Binary/Construct/Composition.agda]: Done

File: src/Relation/Binary/Construct/Composition.agda (hunk 1 above)
Comment (calintat) [reply src/Relation/Binary/Construct/Composition.agda]: Done

**Comment by MatthewDaggitt** [comment]
Thanks! I'll leave it up for a few more days in case anyone else has any comments and then I'll merge it in.

**Comment by calintat** [comment]
Awesome, thanks!

**Pull request merged**